### PR TITLE
Change default Kubernetes version to v1.12.8

### PR DESCRIPTION
**Pull request by dlipovetsky**

### CODE REVIEW

File: cmd/workaround_flannel_issue_1044.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/platform9/nodeadm/constants"
+)
+
+// ensureFlannelDaemonSetTolerations patches the flannel daemonset so that it
+// tolerates all NoSchedule taints. See https://github.com/coreos/flannel/issues/1044
+func ensureFlannelDaemonSetToleratesAllNoScheduleTaints() error {
```
Comment (ojmhetar) [inline cmd/workaround_flannel_issue_1044.go]: Why not call `patchFlannelDaemonSet()` directly, is it to mimic the convention of `ensureKubeProxyRespectsHostoverride()`?

**Review by ojmhetar — APPROVED** [review]
LGTM. Maybe out of the scope of this review - look like `FeatureGates` in `constants.go` isn't used anywhere, could be removed.

File: cmd/workaround_flannel_issue_1044.go (hunk 1 above)
Comment (dlipovetsky) [reply cmd/workaround_flannel_issue_1044.go]: Yeah, the former is expected to be idempotent. For the kubeproxy workaround, I needed to make the patch idempotent, but here kubectl makes it idempotent on its own.

**Comment by dlipovetsky** [comment]
Thanks for the review @ojmhetar.

Yes, we need to clean up constants in general.

File: constants/constants.go [hunk 2]
```diff
@@ -8,22 +8,21 @@ import (
...
 	FlannelVersion                        = "v0.10.0"
-	MetalLBVersion                        = "master"
 	DefaultPodNetwork                     = "10.244.0.0/16"
 	DefaultDNSIP                          = "10.96.0.10"
 	DefaultServiceSubnet                  = "10.96.0.0/12"
 	DefaultDNSDomain                      = "cluster.local"
 	DefaultRouterID                       = 42
 	KubeadmConfig                         = "/tmp/kubeadm.yaml"
-	CoreDNSVersion                        = "1.1.3"
+	CoreDNSVersion                        = "1.2.2"
```
Comment (sarun87) [inline constants/constants.go]: Would this switch remove the older coreDNS installation during an upgrade?

**Review by sarun87 — APPROVED** [review]
Looks good to me otherwise. See my question inline.

**Pull request merged**